### PR TITLE
Docs: Edit issue/PR waiting period docs (fixes #6009)

### LINKED ISSUE #6009: Proposal: Leave issues/PRs open for a set amount of time
The team is fairly large and distributed at this point, with people from a lot of different time zones all over the world (United States, Japan, Russia, Turkey...probably more?). Given that, and the different rates at which people can get online to review issues and PRs, I'd like to propose that we adopt the system used by Node.js, which is basically:

1. PRs should remain open for at least 48 hours before merging (72 hours if over the weekend)
2. Issues should remain open for at least 48 hours before closing (72 hours if over the weekend) - this refers to closing because we aren't going to address the issue. Closing for other reasons (because it's a duplicate, because it's a question that's been answered, etc.) wouldn't need to wait.

The goal is to give more members of the team the opportunity to give feedback on PRs and to ensure issues aren't closed prematurely before people have been able to chime in.

We can always make exceptions as necessary, but those should be reserved for high-priority situations (such as converting every rule to a new format, where waiting will create a lot of merge conflicts for other PRs).

Thoughts?

### PR DESCRIPTION
**Pull request by nzakas**
**What issue does this pull request address?**

#6009 

**What changes did you make? (Give an overview)**

Added docs for when issue and PRs can be closed. Also cleaned up some old info that isn't really how we've been doing things.

**Is there anything you'd like reviewers to focus on?**

Review for accuracy.

### CODE REVIEW

File: docs/maintainer-guide/pullrequests.md [hunk 1]
```diff
@@ -52,6 +52,20 @@ We use the "Merge" button to merge requests into the repository. Before merging
 
 Be sure to say thank you to the submitter before merging, especially if they put a lot of work into the pull request.
 
+Team members may merge a pull request immediately if it:
+
+1. Make a small documentation change
```
Comment (platinumazure) [inline docs/maintainer-guide/pullrequests.md]: s/Make/Makes (parallelism)

File: docs/maintainer-guide/pullrequests.md [hunk 2]
```diff
@@ -52,6 +52,20 @@ We use the "Merge" button to merge requests into the repository. Before merging
...
+1. Is a chore
+1. Fixes a block of other work on the repo (build-related, test-related, dependency-related, etc.)
+1. Is an important fix to get into a patch release
+
+Otherwise, team members should observe a waiting period before closing an issue:
```
Comment (platinumazure) [inline docs/maintainer-guide/pullrequests.md]: Should this say "before merging a pull request:"?

**Comment by platinumazure** [comment]
Made some small comments, but this looks good otherwise. Thanks for clarifying the policies!

File: docs/maintainer-guide/pullrequests.md (hunk 2 above)
Comment (nzakas) [inline docs/maintainer-guide/pullrequests.md]: Yeah, copy-paste error.

**Comment by ilyavolodin** [comment]
LGTM.

**Pull request merged**